src/cargo/core/resolver/mod.rs

```diff
@@ -907,15 +907,22 @@ fn build_features<'a>(s: &'a Summary, method: &'a Method)
             }
             None => {
                 let feat = feat_or_package;
+
+                //if this feature has already been added, then just return Ok
                 if !visited.insert(feat) {
-                    bail!("Cyclic feature dependency: feature `{}` depends \
-                           on itself", feat)
+                    return Ok(());
                 }
+
                 used.insert(feat);
                 match s.features().get(feat) {
                     Some(recursive) => {
                         // This is a feature, add it recursively.
                         for f in recursive {
+                            if f == feat {
+                                bail!("Cyclic feature dependency: feature `{}` depends \
+                                        on itself", feat);
+                            }
```

Comment:
This forbids direct dependency of feature on itself, right? Let's add a test case for this then?

Reply:
Yes that's correct. There was an existing test for this [`cyclic_feature()`](https://github.com/rust-lang/cargo/pull/4473/files#diff-40231dbe3cfb4cb45dee09f272e30cb6L451) so I left it as-is.